content/Registration and Authentication in your Django app with dj-rest-auth part 2 - Bruno Michetti.md

```diff
@@ -0,0 +1,283 @@
+```
+
+With this, after a correct sign-up your app will send a verification email to the entered account. But you didn't finish: as dj-rest-auth [documentation](https://dj-rest-auth.readthedocs.io/en/latest/api_endpoints.html#registration) says, after enabling the email verification, you need to add a path to your urls that will be used by the verification view:
+
+```python
+# Your urls file
+...
+from dj_rest_auth.registration.views import VerifyEmailView
+
+...
+
+urlpatterns = [
+    ...
+    path(
+        'dj-rest-auth/account-confirm-email/',
+        VerifyEmailView.as_view(),
+        name='account_email_verification_sent'
+    ),
+    ...
+]
+
+```
+
+If you don't do that, you will get a [NoReverseMatch](https://docs.djangoproject.com/en/3.1/ref/exceptions/#noreversematch) error.  
+Also, let's use an allauth configuration to activate the email account after the user clicks on the link received in the email. Add this to your settings:
+
+```python
+# Your settings file
+...
+ACCOUNT_CONFIRM_EMAIL_ON_GET = True
+```
+
+And add this path to your urls file `before` the definition of the `dj-rest-auth` registration path:
+
+```python
+# Your urls file
+...
+from dj_rest_auth.registration.views import VerifyEmailView, ConfirmEmailView
+
+...
+
+urlpatterns = [
+    ...
+    path(
+        'dj-rest-auth/registration/account-confirm-email/<str:key>/',
+        ConfirmEmailView.as_view(),
+    ), # Needs to be defined before the registration path
+    path('dj-rest-auth/registration/', include('dj_rest_auth.registration.urls')),
+    path('dj-rest-auth/account-confirm-email/', VerifyEmailView.as_view(), name='account_email_verification_sent'),
+    ...
+]
+
+```
+
+That path needs to be there to avoid a template error in dj-rest-auth. As mentioned in the previous part, dj-rest-auth it's an open source software and sometimes we can find some errors. The good part of this is that anyone can address the discovered bugs.  
+Having this configuration, the user will be redirected to the login page after clicking in the received link. Don't forget to set the `LOGIN_URL` parameter in your settings, otherwise you can get an error if the default value is not a valid URL of your app. During development, you can set the `LOGIN_URL` as the login endpoint; remember that dj-rest-auth has browsable endpoints thanks to django-rest-framework. So you can test sign-up flow by setting for example:
+
+```python
+# Your settings file
+...
+LOGIN_URL = 'http://localhost:8000/dj-rest-auth/login'
+```
+
+Now we need to specify to Django the email backend that is in charge of sending the emails. To locally test the signup feature during development, you can set:
+
+```python
+# Your settings file
+...
+EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
+
+```
+
+In this case the email won't be sent, it will be printed in the console instead. By default, you will see something like this after a successful registration:
```

Comment:
The email, at least from a django point of view will be sent. maybe we can rephrase this.